statistics/src/main/java/com/duckduckgo/app/statistics/store/TimeProvider.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.statistics.store
+
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import java.time.Instant
+import javax.inject.Inject
+
+interface TimeProvider {
```

Comment:
There's already `CurrentTimeProvider` in `common-utils`, would it be possible to agree on a common way to get a timestamp (`Instant.now` vs `SystemClock.elapsedRealtime` vs `System.currentTimeMillis`) and not have different pixels getting timestamps from different classes?

Reply:
I actually looked for existing time provider implementations before adding this one and noticed that there is `CurrentTimeProvider` in `common-utils`. I didn't use it, because it doesn't return timestamp in the conventional sense (a specific point on the timeline), but time since the system has booted, that can't be converted into a date.

While `Instant.now()` and `System.currentTimeMillis()` are based on the same clock, and the difference between those methods comes down to return type, `SystemClock.elapsedRealtime()` serves a different purpose.

I agree that we should have a standarized way obtaining time (that covers both wall clock and time since boot), and not have to create some kind of `TimeProvider` interface in every module, but I left it out of scope of this PR.

Reply:
Right, my bad, in my case all options were interchangeable since I wanted to compute the difference between 2 timestamps, but of course they serve different purposes